File: src/eQuantic.UI.Runtime/src/shared/components/Toast.ts
import { Box, BoxStyle, BuildContext, Column, CornerRadii, EdgeInsets, Overlay, Presence, Pressable, Radius, Row, SizeValue, Space, StatelessComponent, Text } from "../runtime-exports";

export class Toast extends StatelessComponent {
    declare message: string;
    declare status: string;
    declare actionLabel: string;
    declare onAction: () => void;
    constructor(message?: any, status: any = 'info', actionLabel: any = null, onAction: any = null, props?: any) {
        super(props);
        if (message !== undefined) this.message = message;
        if (status !== undefined) this.status = status;
        if (actionLabel !== undefined) this.actionLabel = actionLabel;
        if (onAction !== undefined) this.onAction = onAction;
        if (this.status === undefined) this.status = 'primary';
        this.message = message;this.status = status;this.actionLabel = actionLabel;this.onAction = onAction;
    }

    build(context: BuildContext) {
        let theme = context.theme;let row = new Row(Space.s3, { cross: 'center' });row.add(new Box(new BoxStyle({ width: 8, height: 8, background: theme.colors(this.status).base, cornerRadius: new CornerRadii(Radius.full) })));row.add(new Text(this.message, 'bodyM', theme.textInverse, 2));let label; if (((this.actionLabel != null) && (label = this.actionLabel, true))) {row.add(new Pressable(new Box(new BoxStyle({ padding: EdgeInsets.symmetric(Space.s2, Space.s1) }), new Text(label, 'label', theme.textInverse)), this.onAction, { label: label }));}let pill = new Box(new BoxStyle({ background: theme.textPrimary, cornerRadius: new CornerRadii(theme.shape('full')), elevation: 3, padding: EdgeInsets.symmetric(Space.s4, Space.s3), maxWidth: 480 }), row);let anchor = new Column(0, { width: SizeValue.fill, height: SizeValue.fill, main: 'end', cross: 'center', padding: EdgeInsets.symmetric(Space.s4, Space.s6) });anchor.add(new Presence(pill, 'slideUp'));return new Overlay(anchor, { modal: false });
    }

}

File: src/eQuantic.UI.Components/Toast.cs
using eQuantic.UI.Primitives;

namespace eQuantic.UI.Components;

/// <summary>
/// The design system's Toast/Snackbar (spec C3): a TRANSIENT, NON-MODAL notice anchored to the
/// bottom — an inverse-surface pill (TextPrimary fill / TextInverse label, the pair that flips
/// correctly in both modes), E3, Radius.Full, with an optional single action. DECLARATIVE presence
/// like the Dialog: the app shows it by building it and hides it by not building it — the
/// auto-dismiss TIMER is the app's (or the host clock's) concern, not the component's. The layer is
/// non-modal: everything behind stays clickable; only the pill (and its action) takes input.
/// v1 fences: enter/exit motion (state-transition system) and toast queueing (one at a time).
/// </summary>
public sealed class Toast : StatelessComponent
{
    public Toast(string message, Variant status = Variant.Info,
        string? actionLabel = null, Action? onAction = null)
    {
        Message = message;
        Status = status;
        ActionLabel = actionLabel;
        OnAction = onAction;
    }

    public string Message { get; init; }
    public Variant Status { get; init; }
    public string? ActionLabel { get; init; }
    public Action? OnAction { get; init; }

    public override VisualNode Build(ComponentContext context)
    {
        var theme = context.Theme;

        var row = new Row(gap: Space.S3) { Cross = CrossAlign.Center };
        // The status dot: the one splash of role color on the inverse pill.
        row.Add(new Box(new BoxStyle
        {
            Width = 8, Height = 8,
            Background = theme.Colors(Status).Base,
            CornerRadius = new CornerRadii(Radius.Full),
        }));
        row.Add(new Text(Message, TypeRole.BodyM, theme.TextInverse, maxLines: 2));
        if (ActionLabel is { } label)
        {
            row.Add(new Pressable(new Box(new BoxStyle
            {
                Padding = EdgeInsets.Symmetric(Space.S2, Space.S1),
            }, new Text(label, TypeRole.Label, theme.TextInverse)), OnAction)
            {
                Label = label,
            });
        }

        var pill = new Box(new BoxStyle
        {
            Background = theme.TextPrimary,
            CornerRadius = new CornerRadii(theme.Shape(ShapeScale.Full)),
            Elevation = 3,
            Padding = EdgeInsets.Symmetric(Space.S4, Space.S3),
            MaxWidth = 480,
        }, row);

        var anchor = new Column(gap: 0)
        {
            Width = SizeValue.Fill,
            Height = SizeValue.Fill,
            Main = MainAlign.End,
            Cross = CrossAlign.Center,
            Padding = EdgeInsets.Symmetric(Space.S4, Space.S6),
        };
        // Enter motion (spec §06): the pill rises into place (SlideUp) — the toast entrance.
        anchor.Add(new Presence(pill, PresenceMotion.SlideUp));

        return new Overlay(anchor) { Modal = false };
    }
}
